daxa/mission/erosita.py

```diff
@@ -0,0 +1,703 @@
+        open(field_dir + "{f}.tar.gz".format(f=field), "wb").write(r.content)
+        # unzipping the tar file
+        tarname = field_dir + "{f}.tar.gz".format(f=field)
+        tar = tarfile.open(tarname, "r:gz")
+        tar.extractall(field_dir)
+        tar.close()
+        os.remove(tarname)
+
+        return None
+    
+    def _directory_formatting(self):
+        """
+        Internal method to rearrange the downloaded files from field names into the Obs_ID top layer 
+        directory structure for consistency with other missions in DAXA. To be called after the initial 
+        download of the fields has been completed.
+        """
+        # Only executing the method if new data has been downloaded
+        if os.path.exists(os.path.join(self.raw_data_path, "temp_download")):
+            # Creating a dictionary that stores obs_ids as keys and their field name as values
+            field_dict = self._field_dict_generator()
+
+            # Moving the eventlist for each obs_id from its downloaded path to the path DAXA expects
+            for o in self.filtered_obs_ids:
+                if not os.path.exists(self.raw_data_path + '{o}'.format(o=o)):
+                    os.makedirs(self.raw_data_path + '{o}'.format(o=o))
+                    # The path to the obs_id directory
+                    obs_dir = os.path.join(self.raw_data_path, '{o}'.format(o=o))
+                    # The field the obs_id was downloaded with
+                    field_name = field_dict[o]
+                    # The path to where the obs_id was initially downloaded
+                    field_dir = os.path.join(self.raw_data_path, "temp_download", field_name)
+                    # Not including hidden files in this list
+                    all_files = [f for f in os.listdir(field_dir) if not f.startswith('.')]
+                    # Some fields are in a folder, some are just the files not in a folder
+                    # If they are in a folder, there will only be one file in all files
+                    if len(all_files) == 1:
+                        second_field_dir = all_files[0]
+                        # redefining all_files so it lists the files in the folder
+                        all_files = os.listdir(os.path.join(field_dir, second_field_dir))
+                        # redefining field_dir so in the later block, the source is correct
+                        field_dir = os.path.join(field_dir, second_field_dir)
+                        
+                    # Selecting the eventlist for the obs_id
+                    obs_file_name =  [obs_file for obs_file in all_files if o in obs_file and "eRO" not in obs_file ][0]
+                    source = os.path.join(field_dir, obs_file_name)
+                    dest = os.path.join(obs_dir, obs_file_name)
+                    shutil.move(source, dest)
+
+            # Deleting temp_download directory containing the extra files that were not the obs_id eventlists
```

Comment:
What files are you deleting? Some of them might be necessary for further processing and cleaning, although you know better than me on that front.